src/docfx.website.themes/common/common.js

```diff
@@ -103,36 +102,70 @@ function normalizeGitUrlToHttps(repo) {
 }
 
-function getNewFileUrl(uid, gitContribute, gitUrlPattern) {
+function getNewFileUrl(item, gitContribute, gitUrlPattern) {
     // do not support VSO for now
-    if (!gitContribute || !gitContribute.repo) return '';
-    var repo = gitContribute.repo;
-    if (repo.substr(-4) === '.git') {
-        repo = repo.substr(0, repo.length - 4);
+    if (!item.source) {
+        return '';
     }
 
-    var patternName = getPatternName(repo, gitUrlPattern);
+    var modelGitInfo = getGitContributeFallbackWithGitRemote(item.source.remote, gitContribute);
+    if (!modelGitInfo.repo || !modelGitInfo.branch || !modelGitInfo.path) {
+        return '';
+    }
+
+    if (modelGitInfo.repo.substr(-4) === '.git') {
+        modelGitInfo.repo = modelGitInfo.repo.substr(0, modelGitInfo.repo.length - 4);
+    }
+
+    var patternName = getPatternName(modelGitInfo.repo, gitUrlPattern);
     if (!patternName) return patternName;
-    return gitUrlPatternItems[patternName].generateNewFileUrl(repo, gitContribute.branch, gitContribute.path, uid);
+    return gitUrlPatternItems[patternName].generateNewFileUrl(modelGitInfo, item.uid);
 }
 
 function getRemoteUrl(remote, startLine, gitContribute, gitUrlPattern) {
-    var repo = undefined;
-    var branch = undefined;
-    if (gitContribute && gitContribute.repo) repo = gitContribute.repo;
-    if (repo == undefined && remote && remote.repo) repo = remote.repo;
-    if (gitContribute && gitContribute.branch) branch = gitContribute.branch;
-    if (branch == undefined && remote && remote.branch) branch = remote.branch;
-
-    if (repo == undefined || branch == undefined) return '';
+    var modelGitInfo = getGitContributeFallbackWithGitRemote(remote, gitContribute);
+    if (!modelGitInfo.repo || !modelGitInfo.branch || !modelGitInfo.path) {
+        return '';
+    }
 
-    if (repo.substr(-4) === '.git') {
-        repo = repo.substr(0, repo.length - 4);
+    if (modelGitInfo.repo.substr(-4) === '.git') {
+        modelGitInfo.repo = modelGitInfo.repo.substr(0, modelGitInfo.repo.length - 4);
     }
 
-    var patternName = getPatternName(repo, gitUrlPattern);
+    var patternName = getPatternName(modelGitInfo.repo, gitUrlPattern);
     if (!patternName) return '';
-    return gitUrlPatternItems[patternName].generateUrl(repo, branch, remote.path, startLine);
+
+    modelGitInfo.startLine = startLine;
+    return gitUrlPatternItems[patternName].generateUrl(modelGitInfo);
+}
+
+function getGitContributeFallbackWithGitRemote(gitRemote, gitContribute) {
+    var modelGitInfo = {};
+
+    if (gitContribute && gitContribute.repo) {
+        modelGitInfo.repo = gitContribute.repo;
+    } else if (gitRemote && gitRemote.repo) {
+        modelGitInfo.repo = gitRemote.repo;
+    }
+
+    if (gitContribute && gitContribute.branch) {
+        modelGitInfo.branch = gitContribute.branch;
+    } else if (gitRemote && gitRemote.branch) {
+        modelGitInfo.branch = gitRemote.branch;
+    }
+
+    if (gitContribute) {
```

Comment:
nice catch

---
In reply to: [91652863](https://github.com/dotnet/docfx/pull/979#discussion_r91652863) [](ancestors = 91652863)